Packet++/src/TcpReassembly.cpp

```diff
@@ -132,20 +132,28 @@ void TcpReassembly::reassemblePacket(Packet& tcpData)
 	else if (tcpData.isPacketOfType(IPv6))
 		ipLayer = (Layer*)tcpData.getLayerOfType<IPv6Layer>();
 
-	if (ipLayer == NULL)
+	if (ipLayer == NULL) 
+	{
+		status = Error_NonIpPacket;
 		return;
+	}
+
 
 	// Ignore non-TCP packets
 	TcpLayer* tcpLayer = tcpData.getLayerOfType<TcpLayer>(true); // lookup in reverse order
 	if (tcpLayer == NULL)
+	{
+		status = Error_NonTcpPacket;
 		return;
+	}
 
 	// Ignore the packet if it's an ICMP packet that has a TCP layer
 	// Several ICMP messages (like "destination unreachable") have TCP data as part of the ICMP message.
 	// This is not real TCP data and packet can be ignored
 	if (tcpData.isPacketOfType(ICMP))
 	{
 		LOG_DEBUG("Packet is of type ICMP so TCP data is probably  part of the ICMP message. Ignoring this packet");
+		status = Ignore_IcmpPacket;

```

Comment:
I don't think we need a special enum for that. It's just another case of non-TCP packet